backend/server.js

```diff
@@ -1,24 +1,112 @@
 import cors from "cors";
+import crypto from "crypto";
+import bcrypt from "bcrypt";
 import express from "express";
 import mongoose from "mongoose";
+import expressListEndpoints from "express-list-endpoints";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+// const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/auth";
+const mongoUrl = "mongodb://localhost/auth";
 mongoose.connect(mongoUrl);
 mongoose.Promise = Promise;
 
+const User = mongoose.model("User", {
+  name: {
+    type: String,
+    unique: true,
+  },
+  email: {
+    type: String,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString("hex"),
+  },
+});
+
+// Middleware function
+const authenticateUser = async (req, res, next) => {
+  const user = await User.findOne({ accessToken: req.header("Authorization") });
+  if (user) {
+    req.user = user;
+    next();
+  } else {
+    res.status(401).json({ loggedOut: true });
+  }
+};
+
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Documentation endpoint
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  const documentation = {
+    Welcome: "This is the Authentication API!",
+    Endpoints: {
+      "/": "Get API documentation",
+      "/users": {
+        POST: "Create a new user",
+      },
+      "/sessions": {
+        POST: "Authenticate a returning user",
+      },
+      "/secrets": {
+        GET: "Get secret content (requires authentication)",
+      },
+    },
+  };
+  res.json(documentation);
+});
```

Comment:
you're not really listing the endpoints urls, try with:
```javascript
app.get("/", (req, res) => {
  const endpoints = expressListEndpoints(app);
  const documentation = {
    Welcome: "This is the Authentication API!",
    Endpoints: endpoints.map((endpoint) => ({
      method: endpoint.methods.join(", "),
      path: endpoint.path
    }))
  };
  res.json(documentation);
});
```

Reply:
Sorry Antonella, I forgot to tell that I couldn't finish the auth project. I got stuck first on getting the endpoints to work on postman and later trying to deploy to Netlify, so I had no time to connect the FE to the BE. I will try to work on it in another moment. Just now I'm working on the FP.